### PR TITLE
Add support for Security Advisories Request CVE endpoint

### LINKED ISSUE #2855: Support new API to Request a CVE for a repository security advisory
GitHub Blog post: https://github.blog/changelog/2023-08-03-request-a-cve-identifier-for-your-repository-security-advisory-via-rest-api/
Rest API docs: https://docs.github.com/en/rest/security-advisories/repository-advisories?apiVersion=2022-11-28#request-a-cve-for-a-repository-security-advisory

This would be a great PR for any new contributor to this repo or a new Go developer.
All contributions are greatly appreciated!

Feel free to volunteer for any issue and the issue can be assigned to you so that others don't attempt to duplicate the work.

Please check out our [CONTRIBUTING.md](https://github.com/google/go-github/blob/master/CONTRIBUTING.md) guide to get started. (In particular, please remember to `go generate ./...` and don't use force-push to your PRs.)

Thank you!

### PR DESCRIPTION
**Pull request by be0x74a**
Fixes #2855

### CODE REVIEW

File: README.md
Comment (gmlewis) [inline README.md:399]: Please revert these changes, as they are controlled by maintainers of this repo only.

File: github/github.go
Comment (gmlewis) [inline github/github.go:31]: Please revert this line.

File: github/security_advisories.go
Comment (gmlewis) [inline github/security_advisories.go:15]:
```suggestion
// RequestCVE requests a Common Vulnerabilities and Exposures (CVE) for a repository security advisory.
// The ghsaID is the GitHub Security Advisory identifier of the advisory.
```

File: github/security_advisories.go
Comment (gmlewis) [inline github/security_advisories.go:30]: Please remove the blank line that is separating the potential cause of an error with the handling of that error.

```suggestion
	if err != nil {
		if _, ok := err.(*AcceptedError); ok {
```

File: github/security_advisories_test.go
Comment (gmlewis) [inline github/security_advisories_test.go:29]:
```suggestion
	const methodName = "RequestCVE"
	testBadOptions(t, methodName, func() (err error) {
		_, err := client.SecurityAdvisories.RequestCVE(ctx, "\n", "\n", "\n")
		return err
	})

```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @be0x74a

File: README.md
Comment (be0x74a) [reply README.md]: Thanks for the info, reverted

File: github/github.go
Comment (be0x74a) [reply github/github.go]: Reverted

File: github/security_advisories.go
Comment (be0x74a) [reply github/security_advisories.go]: Done

File: github/security_advisories.go
Comment (be0x74a) [reply github/security_advisories.go]: Done, thanks the suggestion

File: github/security_advisories.go
Comment (gmlewis) [reply github/security_advisories.go]: Sorry, this was not completed.  Please add the description shown above for "ghsaID".

File: github/security_advisories.go
Comment (gmlewis) [inline github/security_advisories.go:27]: Please remove this blank line that is separating the potential source of an error with the handling of the same error.

File: github/security_advisories.go
Comment (be0x74a) [reply github/security_advisories.go]: Sorry, for the back and forth. Completely missed this

File: github/security_advisories.go
Comment (be0x74a) [reply github/security_advisories.go]: Done, thanks

**Review by gmlewis — APPROVED** [review]
Thank you, @be0x74a !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by be0x74a** [comment]
> 

Thank you for the time you took to review and feedback on the PR @gmlewis

**Review by vandanrohatgi — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
Thank you, @vandanrohatgi !
Merging.

**Pull request merged**
